lib/nsq/discovery.rb

```diff
@@ -69,8 +69,15 @@ def get_nsqds(lookupd, topic = nil)
 
       uri.query = "ts=#{Time.now.to_i}"
       if topic
-        uri.path = '/lookup'
-        uri.query += "&topic=#{topic}"
+        if topic.include? '#'
+          uri.path = '/lookup'
+          topic_split = topic.split('#')
+          uri.query += "&topic=#{topic_split.first}"
+          uri.query += "%23#{topic_split.last}"
+        else
+          uri.path = '/lookup'
+          uri.query += "&topic=#{topic}"
+        end
```

Comment:
Is nsq always going to decode the topic? if that is the case you could like solve this by simply `URI.escape`ing the topic like this:
```ruby
if topic
  uri.path = '/lookup'
  uri.query += "&topic=#{URI.escape(topic)}"
else
  uri.path = '/nodes'
end
```